### DISCUSSION TITLE
How to add style into current file?

### DISCUSSION THREAD

**Discussion by hxhgxy** [OfficeDev/Open-XML-SDK#1256] [Q&A]
I try to add new styles into stylesheet of current file:

`
		
		private Stylesheet AddNewStyles(Stylesheet stylesheet)
		{
			Stylesheet newStyle = stylesheet;
			Fonts fonts = newStyle.Fonts;
			Fills fills = newStyle.Fills;
			Borders borders = newStyle.Borders;
			CellFormats cellFormats = newStyle.CellFormats;

			UInt32Value fontCount = fonts.Count;
			UInt32Value fillCount = fills.Count;
			UInt32Value borderCount = borders.Count;

			#region 字体
			Font fontTitle = new Font();
			Bold boldTitle = new Bold();
			FontSize fontSizeTitle = new FontSize() { Val = 10D };
			Color colorTitle = new Color() { Rgb = "FFFFFFFF" };
			FontName fontNametTitle = new FontName() { Val = "微软雅黑" };
			FontFamilyNumbering fontFamilyNumberingTitle = new FontFamilyNumbering() { Val = 2 };
			FontCharSet fontCharSetTitle = new FontCharSet() { Val = 134 };

			fontTitle.Append(boldTitle);
			fontTitle.Append(fontSizeTitle);
			fontTitle.Append(colorTitle);
			fontTitle.Append(fontNametTitle);
			fontTitle.Append(fontFamilyNumberingTitle);
			fontTitle.Append(fontCharSetTitle);

			Font fontTable = new Font();
			FontSize fontSizeTable = new FontSize() { Val = 10D };
			Color colorTable = new Color() { Theme = (UInt32Value)1U };
			FontName fontNameTable = new FontName() { Val = "微软雅黑" };
			FontFamilyNumbering fontFamilyNumberingTable = new FontFamilyNumbering() { Val = 2 };
			FontCharSet fontCharSetTable = new FontCharSet() { Val = 134 };

			fontTable.Append(fontSizeTable);
			fontTable.Append(colorTable);
			fontTable.Append(fontNameTable);
			fontTable.Append(fontFamilyNumberingTable);
			fontTable.Append(fontCharSetTable);

			fonts.Append(fontTitle);
			fonts.Append(fontTable);
			#endregion

			#region 填充
			#region 2 灰色填充，表格标题
			Fill fillTitle = new Fill();

			PatternFill patternFillTitle = new PatternFill() { PatternType = PatternValues.Solid };
			ForegroundColor foregroundColor1 = new ForegroundColor() { Rgb = "FF505050" };
			BackgroundColor backgroundColor1 = new BackgroundColor() { Indexed = (UInt32Value)64U };

			patternFillTitle.Append(foregroundColor1);
			patternFillTitle.Append(backgroundColor1);

			fillTitle.Append(patternFillTitle);
			#endregion

			#region 3 浅灰色填充，表格隔行设色
			Fill fillTable = new Fill();

			PatternFill patternFillTable = new PatternFill() { PatternType = PatternValues.Solid };
			ForegroundColor foregroundColor2 = new ForegroundColor() { Rgb = "FFE6E6E6" };
			BackgroundColor backgroundColor2 = new BackgroundColor() { Indexed = (UInt32Value)64U };

			patternFillTable.Append(foregroundColor2);
			patternFillTable.Append(backgroundColor2);

			fillTable.Append(patternFillTable);
			#endregion

			fills.Append(fillTitle);
			fills.Append(fillTable);
			#endregion

			#region 边框

			#region 3 上下左右边框，表格
			Border borderTable = new Border();

			LeftBorder leftBorderTable = new LeftBorder() { Style = BorderStyleValues.Thin };
			Color colorTable1 = new Color() { Rgb = "FFC8C8C8" };

			leftBorderTable.Append(colorTable1);

			RightBorder rightBorderTable = new RightBorder() { Style = BorderStyleValues.Thin };
			Color colorTable2 = new Color() { Rgb = "FFC8C8C8" };

			rightBorderTable.Append(colorTable2);

			TopBorder topBorderTable = new TopBorder() { Style = BorderStyleValues.Thin };
			Color colorTable3 = new Color() { Rgb = "FFC8C8C8" };

			topBorderTable.Append(colorTable3);

			BottomBorder bottomBorderTable = new BottomBorder() { Style = BorderStyleValues.Thin };
			Color colorTable4 = new Color() { Rgb = "FFC8C8C8" };

			borderTable.Append(leftBorderTable);
			borderTable.Append(rightBorderTable);
			borderTable.Append(topBorderTable);
			borderTable.Append(bottomBorderTable);
			#endregion

			borders.Append(borderTable);
			#endregion

			#region 单元格样式
			#region Index 9：表格标题
			CellFormat cellFormatTitle = new CellFormat()
			{
				NumberFormatId = (UInt32Value)0U,
				FontId = (UInt32Value)fontCount,
				FillId = (UInt32Value)fillCount,
				BorderId = (UInt32Value)borderCount,
				FormatId = (UInt32Value)0U,
				ApplyFont = true,
				ApplyFill = true,
				ApplyBorder = true
			};
			Alignment alignmentTitle = new Alignment() { Vertical = VerticalAlignmentValues.Center };

			cellFormatTitle.Append(alignmentTitle);
			#endregion

			#region Index 10：表格内容，奇数行
			CellFormat cellFormatRow1 = new CellFormat()
			{
				NumberFormatId = (UInt32Value)0U,
				FontId = (UInt32Value)fontCount + 1,
				FillId = (UInt32Value)0U,
				BorderId = (UInt32Value)borderCount,
				FormatId = (UInt32Value)0U,
				ApplyFont = true,
				ApplyBorder = true
			};
			Alignment alignmentRow1 = new Alignment() { Vertical = VerticalAlignmentValues.Center };

			cellFormatRow1.Append(alignmentRow1);
			#endregion

			#region Index 11：表格内容，偶数行
			CellFormat cellFormatRow0 = new CellFormat()
			{
				NumberFormatId = (UInt32Value)0U,
				FontId = (UInt32Value)fontCount + 1,
				FillId = (UInt32Value)fontCount + 1,
				BorderId = (UInt32Value)borderCount,
				FormatId = (UInt32Value)0U,
				ApplyFont = true,
				ApplyFill = true,
				ApplyBorder = true
			};
			Alignment alignmentRow0 = new Alignment() { Vertical = VerticalAlignmentValues.Center };

			cellFormatRow0.Append(alignmentRow0);
			#endregion

			cellFormats.Append(cellFormatTitle);
			cellFormats.Append(cellFormatRow1);
			cellFormats.Append(cellFormatRow0);
			#endregion

			return newStyle;
		}
`

What's arong with codes above? Thanks a lot!

**Comment by hxhgxy** [comment]
Shame to find that make "fillCount" into "fontCount" in mistake.

**Reply by tomjebo** [reply]
@hxhgxy I assume you're saying this is the fix to your problem? Can you confirm?

**Reply by hxhgxy** [reply]
> @hxhgxy I assume you're saying this is the fix to your problem? Can you confirm?

yes, thanks.

**Comment by tomjebo** [comment]
@hxhgxy typically this kind of "how-to" question can be asked on StackOverflow.com with a tag of openxml-sdk:
https://stackoverflow.com/questions/tagged/openxml-sdk

Having said that, a typical approach on this kind of problem is to open with Excel, let Excel repair it and compare the differences in the underlying styles.xml part.
